exec/session.go

```diff
@@ -222,6 +222,22 @@ func (s *Session) Must(ctx context.Context, funcv *bigslice.FuncValue, args ...i
 	return res
 }
 
+// Discard discards the storage resources held by the subgraph given by roots.
+// This should be used to discard tasks whose results are no longer needed. If
+// the task results are needed by another computation, they will be recomputed.
+// Discarding is best-effort, so no error is returned.
+func (s *Session) Discard(ctx context.Context, roots []*Task) {
+	var wg sync.WaitGroup
+	iterTasks(roots, func(task *Task) {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			s.executor.Discard(ctx, task)
```

Comment:
I just noticed 79dcd08e091ffa56f1bfb4bb8e663cc2800bff45. I leave it up to your judgement. If messiness is a concern, `traverse` may offer a clean and readable concurrency-limited solution.

Reply:
I did this with a limiter to avoid materializing set of tasks. We still will spin a goroutine for each task, but we limit the number to 64. Let me know if you have objections.